File: js/sample-issue-tracker/src/Issues/Application/IIssueRepository.ts
export interface IIssueRepository {
  getByIdAsync(id: IssueId): Promise<Issue | null>;
  listAsync(): Promise<Issue[]>;
  saveAsync(issue: Issue): Promise<void>;
  existsAsync(id: IssueId): Promise<boolean>;
  listBySprintAsync(sprintKey: string): Promise<Issue[]>;
  searchAsync(status: IssueStatus | null, priority: IssuePriority | null, assigneeId: UserId | null, sprintKey: string | null, page: PageRequest): Promise<PageResult<Issue>>;
}

File: samples/SampleIssueTracker/Issues/Application/IIssueRepository.cs
using SampleIssueTracker.Issues.Domain;
using SampleIssueTracker.SharedKernel;

namespace SampleIssueTracker.Issues.Application;

public interface IIssueRepository
{
    Task<Issue?> GetByIdAsync(IssueId id);

    Task<IReadOnlyList<Issue>> ListAsync();

    Task SaveAsync(Issue issue);

    Task<bool> ExistsAsync(IssueId id);

    Task<IReadOnlyList<Issue>> ListBySprintAsync(string sprintKey);

    Task<PageResult<Issue>> SearchAsync(
        IssueStatus? status,
        IssuePriority? priority,
        UserId? assigneeId,
        string? sprintKey,
        PageRequest page
    );
}
